### PR TITLE
Create KOLIBRI_HOME in userData

### PR DESCRIPTION
**Pull request by danigm**
https://phabricator.endlessm.com/T33468

### CODE REVIEW

File: kolibri-electron/src/index.js [hunk 1]
```diff
@@ -28,7 +29,7 @@ let django = null;
 
 let KOLIBRI_HOME_TEMPLATE = '';
 let KOLIBRI_EXTENSIONS = path.join(__dirname, 'Kolibri', 'kolibri', 'dist');
-let KOLIBRI_HOME = path.join(os.homedir(), `.endless-key-${KEY_VERSION}`);
+let KOLIBRI_HOME = path.join(userData, `endless-key-${KEY_VERSION}`);
```
Comment (wjt) [inline kolibri-electron/src/index.js]: I don't think it is correct to use KEY_VERSION in the directory name. If we do that, we can never update the version name without existing Windows Store users effectively losing their data.

**Review by wjt — CHANGES_REQUESTED** [review]

File: kolibri-electron/src/index.js (hunk 1 above)
Comment (wjt) [reply kolibri-electron/src/index.js]: Do you need to make this conditional on whether it's being run from a USB stick? (Is the KOLIBRI_HOME meant to live on the USB drive or on the computer's hard disk in that case?)

Comment (danigm) [reply kolibri-electron/src/index.js]: In the last Endless Key releases we have the KOLIBRI_HOME in the computer hard disk, so even if it's a USB stick, the home is created in the computer, not in the USB.

Comment (danigm) [reply kolibri-electron/src/index.js]: > I don't think it is correct to use KEY_VERSION in the directory name. If we do that, we can never update the version name without existing Windows Store users effectively losing their data.

Yes, that's a problem that we'll need to solve for the Windows Store option. In the current EK version, the usage of Kolibri is with an anonymous user, so there's "no user data", just the metrics, so it's safe to remove completely the KOLIBRI_HOME when there's a new release.

With this new version and with a different user case, we could want to keep the user data, that could be the progress, maybe favorites in the future?, etc. But to do that we should make sure that the database is updated with new versions correctly and manually update the content. So it's a bit more complex than just wipe the KOLIBRI_HOME and use the new one.

Comment (wjt) [reply kolibri-electron/src/index.js]: Doesn't Kolibri update its own database on startup?

**Review by wjt — APPROVED** [review]

File: kolibri-electron/src/index.js (hunk 1 above)
Comment (danigm) [reply kolibri-electron/src/index.js]: > Doesn't Kolibri update its own database on startup?

Yes, it should work, but if we want more control over the content in the application, we should enforce channel update, but that's something to think about and decide what to do. I'm thinking about removing "bad" content with an update or something like that.

But by default I think it should work correctly, we only need to clear the browser cache on update to make sure that the javascript is updated with latest versions.

**Pull request merged**